### PR TITLE
Add p2p support to nixos service

### PR DESCRIPTION
**Pull request by jbgi**
this switch iohk-nix to this branch: https://github.com/input-output-hk/iohk-nix/pull/474/files

### CODE REVIEW

**Review by coot — APPROVED** [review]

**Review by disassembler — APPROVED** [review]

File: nix/nixos/cardano-node-service.nix [hunk 1]
```diff
@@ -404,6 +458,38 @@ in {
         description = ''Internal representation of the config.'';
       };
 
+      targetNumberOfRootPeers = mkOption {
+        type = types.int;
+        default = cfg.nodeConfig.TargetNumberOfRootPeers or 30;
```
Comment (karknu) [inline nix/nixos/cardano-node-service.nix:463]: I think that a default value of 30 is too few root peers. I propose that it is increased to 60.
Given your formulas below it means that by default the node will have 20 active peers which is what we have assumed.

File: nix/nixos/cardano-node-service.nix [hunk 2]
```diff
@@ -14,13 +14,34 @@ let
...
-      Producers = cfg.producers ++ (cfg.instanceProducers i);
-    });
+    topology = if cfg.topology != null then cfg.topology else toFile "topology.yaml" (toJSON ({
+      LocalRoots = {
+        groups = map (g: {
+          localRoots = {
+            inherit (g) addrs;
+            advertise = g.advertise or false;
+          };
+          valency = g.valency or 1;
```
Comment (jbgi) [inline nix/nixos/cardano-node-service.nix:33]:
```suggestion
          valency = g.valency or (length g.addrs);
```

**Pull request merged**